src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannel.cs

```diff
@@ -18,7 +18,7 @@
 namespace System.ServiceModel.Channels
 {
     // This class is sealed because the constructor could call Abort, which is virtual
-    internal sealed class ServiceChannel : CommunicationObject, IChannel, IClientChannel, IDuplexContextChannel, IOutputChannel, IRequestChannel, IServiceChannel
+    internal /*sealed*/ class ServiceChannel : CommunicationObject, IChannel, IClientChannel, IDuplexContextChannel, IOutputChannel, IAsyncRequestChannel, IServiceChannel
```

Comment:
Why are the methods virtual? They come from IAsyncRequestChannel so there's no requirement for them to be virtual.

---
In reply to: [78188771](https://github.com/dotnet/wcf/pull/1515#discussion_r78188771) [](ancestors = 78188771,78154483)